pkg/pulumiyaml/codegen/gen_program.go

```diff
@@ -222,12 +222,10 @@ func (g *generator) genResourceOpts(opts *pcl.ResourceOptions) *syn.ObjectNode {
 
 func (g *generator) genResource(n *pcl.Resource) {
 	properties := make([]syn.ObjectPropertyDef, len(n.Inputs))
-	var additionalSecrets []*syn.StringNode
 	for i, input := range n.Inputs {
 		value := input.Value
 		if f, ok := value.(*model.FunctionCallExpression); ok && f.Name == "secret" {
-			contract.Assert(len(f.Args) == 1)
-			additionalSecrets = append(additionalSecrets, syn.String(input.Name))
+			contract.Assertf(len(f.Args) == 1, "Expected exactly one argument to secret, got %d", len(f.Args))
 			value = f.Args[0]
 		}
```

Comment:
I imagine this whole branch can go. Just let the secret function be eval'd by g.expr below.

Reply:
I think that'll error out because `generator.function` doesn't recognize the `"secret"` function.
That's fixable but I don't think this PR should do that.

Reply:
Ah... that's probably why this was _supposed_ to be doing something with additionalSecrets.

Well this is no more broken than it was before, we'll have to fix this somehow later.

Reply:
Ack. Created https://github.com/pulumi/pulumi-yaml/issues/488 to track follow up